source/write/bulk-write.txt

```diff
@@ -0,0 +1,343 @@
+
+Overview
+--------
+
+This guide shows you how to use the {+driver-short+} to perform a bulk
+write operation that makes multiple changes to your data in a single
+database call.
+
+Consider a situation that requires you to insert documents, update
+documents, and delete documents for the same task. If you use
+the individual write methods to perform each type of operation, each write
+accesses the database separately. You can use a bulk write operation to
+optimize the number of calls your application makes to the server.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your Scala application, create a ``MongoClient`` that connects to an Atlas cluster
+and assign the following values to your ``database`` and ``collection`` variables:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :language: scala
+   :dedent:
+   :start-after: start-db-coll
+   :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Define the Write Operations
+---------------------------
+
+For each write operation you want to perform, create a corresponding
+instance of one of the following operation classes that inherit from the
+generic ``WriteModel`` class:
+
+- ``InsertOneModel``
+- ``UpdateOneModel``
+- ``UpdateManyModel``
+- ``ReplaceOneModel``
+- ``DeleteOneModel``
+- ``DeleteManyModel``
+
+Then, pass a list of these instances to the ``bulkWrite()`` method.
+
+The following sections show how to create and use instances of the
+preceding classes. The :ref:`scala-bulkwrite-method` section
+demonstrates how to pass a list of models to the ``bulkWrite()`` method
+to perform the bulk operation.
+
+Insert Operations
+~~~~~~~~~~~~~~~~~
+
+To perform an insert operation, create an ``InsertOneModel`` instance and specify
+the document you want to insert.
+
+The following example creates an instance of ``InsertOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-insert-one
+   :end-before: end-bulk-insert-one
+   :language: scala
+   :copyable:
+   :dedent:
+
+To insert multiple documents, create an instance of ``InsertOneModel``
+for each document.
+
+.. important::
+
+   When performing a bulk operation, the ``InsertOneModel`` cannot
+   insert a document with an ``_id`` that already exists in the
+   collection. In this situation, the driver throws a
+   ``MongoBulkWriteException``.
+
+Update Operations
+~~~~~~~~~~~~~~~~~
+
+To update a document, create an instance of ``UpdateOneModel`` and pass
+the following arguments:
+
+- A **query filter** that specifies the criteria used to match documents in your collection
+- The update operation you want to perform. For more information about update
```

Comment:
S: Remove the articles at the beginning of each  [list item](https://www.mongodb.com/docs/meta/style-guide/style/lists/#list-items); also, since the second bullet point has periods I'd add one to the first:

```suggestion
- **Query filter** that specifies the criteria used to match documents in your collection.
- Update operation you want to perform. For more information about update
```